### PR TITLE
fix: media shouldn't open permissions dialog

### PR DESCRIPTION
**Pull request by indutny**
#### Description of Change

Playing media shouldn't open Accessibility permissions dialog on macOS.
However, we still need to watch for media events, just not globally and
`media_keys_listener_` is an API over global capture of the media keys.

The fix is to let chromium call `UpdateWhichKeysAreListenedFor` which
will call `UpdateSystemMediaControlsEnabledControls` and watch for
events on `system_media_controls_` without triggering permissions popup.

Fix: #31558

cc @codebytere 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where playing media would open Accessibility permissions dialog on macOS.

### CODE REVIEW

**Comment by indutny** [comment]
FWIW, I tested it on the test case from https://github.com/electron/electron/pull/31492 and our app. Without use of `globalShortcut` the playback menu on macOS looks like it did in #31492:

image

and with `globalShortcut` I get accessibility dialog and the shortcut is working.

**Comment by indutny** [comment]
cc @zcbenz @MarshallOfSound @VerteDinde @jkleinsc @indutny-signal

**Comment by indutny** [comment]
It'd be great if backport to 15-x-y could be released with this change. Our weekly release pipeline is currently blocked on because of this issue :sob:

File: patches/chromium/fix_media_key_usage_with_globalshortcuts.patch
Comment (indutny) [inline patches/chromium/fix_media_key_usage_with_globalshortcuts.patch:84]: I have a feeling that this define wasn't even working because it is `OS_MAC` :joy:

**Comment by indutny-signal** [comment]
I think I messed something up while submitting the patch. Please give me few moments to figure it out. Sorry for the noise!

**Comment by indutny** [comment]
Added update to the patches as suggested by CircleCI.

**Comment by indutny** [comment]
FWIW, 13-x-y might need the back port too. We don’t use it at our company anymore, but the issue is present there as well.

thanks!

**Review by VerteDinde — APPROVED** [review]
Thanks for the PR, @indutny!

I pulled the changes locally and tested - this patch change looks good to me, but I'd like a second opinion from another maintainer with a little more experience with the media APIs before we merge 👍

File: patches/chromium/fix_media_key_usage_with_globalshortcuts.patch
Comment (indutny-signal) [inline patches/chromium/fix_media_key_usage_with_globalshortcuts.patch:76]: `media_keys_listener_` is implemented by `ui/base/accelerators/media_keys_listener_mac.mm` and it provides API for listening for media keys *globally*. In fact, the code in `media_kes_listener_mac.mm` calls `CGEventTapCreate` whenever new media key is watched and this immediately causes an Accessibility popup to appear since this permission is required for `CGEventTapCreate` to function at all.

File: patches/chromium/fix_media_key_usage_with_globalshortcuts.patch
Comment (indutny-signal) [inline patches/chromium/fix_media_key_usage_with_globalshortcuts.patch:96]: The trick thus is to let `system_media_controls_` handle the keystroke whenever we have internal handling enabled (i.e. when there are no global accelerators assigned for a media key in electron), so we check `media_key_handling_enabled_` above and fall through here if it is `false` (meaning that there are no accelerators in electron).

**Comment by indutny-signal** [comment]
Thanks @VerteDinde ! I added a few comments to clarify what I've done in the patch. Hope it helps!

**Review by zcbenz — APPROVED** [review]

**Comment by indutny-signal** [comment]
Thanks for reviews, everyone!

@codebytere is there a chance you could take a look at this? You've submitted the original fix so I'd appreciate a review from you a lot 🤗

**Review by codebytere — APPROVED** [review]

**Comment by indutny-signal** [comment]
Fantastic, thanks @codebytere ! Please let me know if there's any additional action that I should take before this commit can be merged and backported. I'll be happy to help in any way (in part because we wish to get the update with the next 15-x-y release :grin:)

**Comment by indutny-signal** [comment]
Yaaay! Thank you!

**Comment by ArmelChesnais** [comment]
Could this be backported to 14? 14.2.1 displays the same issue.

**Pull request merged**
